### PR TITLE
Move login related data away from app object

### PR DESCRIPTION
**Pull request by masvelio**
## Link to Issue
Closes: #8297 

## Description of Changes
- removed few login-related references from app object
  - `loginState` - unnecessary property that was increasing the complexity. Was depending on the presence of user object returned from the `/status` route ([see here](https://github.com/hicommonwealth/commonwealth/pull/8815/files#diff-f62e3e0234a1d6f2265aa43c85590a6261d39615498a10ee9dca909461770dddR178))
  - `loginStateEmitter` - legacy approach from post mithril time. Removed because rerenders work fine on those scenarios 
  - `loginStatusLoaded` - syntactic sugar for loginState. Also was used for showing loader but I tested it out and in my opinion it is not useful anymore, so it got deleted
  - `isLoggedIn` - moved to user.isLoggedIn. Because it is zustand state, every component that use userStore is auto updated when the login state is being changed. We were waiting for this change since the very beginning of our react transition

## Test Plan
- smoke testing of the app, many files changes but it is basically about showing on/off logic depending on user being logged in/out

## Deployment Plan

1. n/a

## Other Considerations

- n/a

### CODE REVIEW

**Review by jnaviask — APPROVED** [review]
Really fantastic work!

File: packages/commonwealth/client/scripts/state/index.ts [hunk 1]
```diff
@@ -151,7 +133,6 @@ declare const window: any;
 // On logout: called to reset everything
 export async function initAppState(
   updateSelectedCommunity = true,
-  shouldRedraw = true,
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/state/index.ts:154]: 🐐

File: packages/commonwealth/client/scripts/views/pages/overview/index.tsx [hunk 2]
```diff
@@ -31,14 +29,6 @@ const OverviewPage = () => {
     // TODO: ask for a pinned thread prop here to show pinned threads
   });
 
-  useEffect(() => {
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/overview/index.tsx:34]: 🔥

File: packages/commonwealth/client/scripts/views/modals/AuthModal/useAuthentication.tsx [hunk 3]
```diff
@@ -402,7 +403,7 @@ const useAuthentication = (props: UseAuthenticationProps) => {
       // @ts-expect-error StrictNullChecks
       await verifySession(session);
       // @ts-expect-error <StrictNullChecks>
-      await onLogInWithAccount(account, false, true, false);
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/modals/AuthModal/useAuthentication.tsx:405]: not sure why we removed the last boolean?

**Review by mzparacha — APPROVED** [review]
Amazing, loved to see us finally getting rid of `app.loginStateEmitter.on('redraw'....` 🔥

File: packages/commonwealth/client/scripts/views/modals/AuthModal/useAuthentication.tsx (hunk 3 above)
Comment (masvelio) [reply packages/commonwealth/client/scripts/views/modals/AuthModal/useAuthentication.tsx:405]: last argument is `shouldRedrawApp`. It was removed whatsoever

**Pull request merged**